SU-HSDO/suhumsci
# Fix(STN-937):  Fix menu active links using front token with hb-dark-pattern

# READY FOR REVIEW

## Summary
When using the Dark Inversion version of the masthead, and also using the the Drupal token for the home page of `<front>` for the Home menu link, the menu link was appearing wrong when the link was on that page as active.

![image](https://user-images.githubusercontent.com/9355698/147299728-ff6c557c-0f5d-411b-a941-ad23334c2ad6.png)

Upon investigation it was found that when using a `node/1` path for home the link was getting the `hb-main-nav-link--active-trail` class added to the link, which was adding a background color to the active link class.

When using the Drupal token, a path is not followed and that class is not generated by the menu on the link itself. Therefore only changing the color of the text for the Dark Inversion of the menu but not giving it the background that would make the active link accessible.

_This work adds a background color to all active links in the menu within the Dark Inversion on the first level, there are also classes that cover the lvl2, etc, these remained in the styles because the specificity was there because of previous background color bugs on the lower levels. (There is likely room for refactoring here, but would also affect the non-inverted masthead links as well. Which may be more scope and testing than is meant for this card.)_

JIRA Card: https://sparkbox.atlassian.net/browse/STN-937

## Need Review By (Date)
1/3

## Urgency
medium

## Steps to Test
1. npm run test
2. Ensure that your Theme Settings [here](https://sparkbox-sandbox.suhumsci.loc/admin/appearance/settings) are set to use the Dark Masthead for whichever theme you are checking first, both will need to be tested.
2. Edit the Main Menu link for Home [here](https://sparkbox-sandbox.suhumsci.loc/admin/structure/menu/item/1/edit), change the link to `<front>`.
3. Save the link/menu.
4. Go to the Home page and resize your browser to utilize the mobile menu.
5. The Home link should be active and look similar to this:
![image](https://user-images.githubusercontent.com/9355698/147299441-3c812764-ef01-438c-8630-e4391c575dcc.png)

## Verification Checklist
- [x] Home link now is accessible when active with the light background when using Dark Masthead for the menu in mobile.
- [ ] Home and other Active links within the desktop menu using the Dark Masthead are unaffected.
- [x] Active Links in the lower levels are still styled accurately within the Dark Masthead mobile menu.
- [ ] Home and other Active links within the desktop menu using the Light Masthead are unaffected.
- [x] This styling is accurate on both Traditional and Colorful themes.
- [ ] This styling changes accurately and remains accessible between color pairings on each theme.


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
